### ISSUE TITLE
/system/lt.h:48:29: error: cast between incompatible function types from 

### ISSUE THREAD

**Issue by rx80** [tsoding/nothing#910]
OS: Linux, gcc version 9.1.0
```
In file included from .../nothing/src/game/level/level_editor/player_layer.h:6,
                 from .../nothing/src/game/level/level_editor.c:10:
.../nothing/src/game/level/level_editor.c: In function 'level_editor_dump':
.../nothing/src/system/lt.h:48:29: error: cast between incompatible function types from 'int (*)(FILE *)' {aka 'int (*)(struct _IO_FILE *)'} to 'void (*)(void *)' [-Werror=cast-function-type]
   48 |     lt_push(lt, (void*)res, (Dtor)dtor)
      |                             ^
.../nothing/src/game/level/level_editor.c:268:22: note: in expansion of macro 'PUSH_LT'
  268 |     FILE *filedump = PUSH_LT(
      |                      ^~~~~~~
cc1: all warnings being treated as errors

```

**Comment by rexim** [comment]
@rx80 What is OS? What is compiler? What is compiler's version?

**Comment by RIscRIpt** [comment]
I can confirm.
I got this error as well.

**Comment by rexim** [comment]
I guess the recent GCC became a little bit more strict. We had a wrapper around `fclose` to prevent something like this in the past, but after the Lt rewrite we deleted the wrapper. We should probably bring it back.

**Comment by rexim** [comment]
@RIscRIpt @rx80 @apa420 Should be fixed in #934. Please check it out when you get a chance.

**Comment by RIscRIpt** [comment]
Works for me.
